hikari/interactions/modal_interactions.py

```diff
@@ -0,0 +1,343 @@
+# -*- coding: utf-8 -*-
+# cython: language_level=3
+# Copyright (c) 2020 Nekokatt
+# Copyright (c) 2021 davfsa
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Models and enums used for Discord's Modals interaction flow."""
+
+from __future__ import annotations
+
+__all__: typing.List[str] = [
+    "ModalResponseTypesT",
+    "ModalInteraction",
+    "PartialTextInput",
+    "ModalInteraction",
+]
+
+import typing
+
+import attr
+
+from hikari import channels
+from hikari import guilds
+from hikari import messages
+from hikari import snowflakes
+from hikari import traits
+from hikari.interactions import base_interactions
+from hikari.internal import attr_extensions
+
+if typing.TYPE_CHECKING:
+    from hikari import users as _users
+    from hikari.api import special_endpoints
+
+ModalResponseTypesT = typing.Literal[
+    base_interactions.ResponseType.MESSAGE_CREATE, 4, base_interactions.ResponseType.DEFERRED_MESSAGE_CREATE, 5
+]
+"""Type-hint of the response types which are valid for a modal interaction.
+
+The following types are valid for this:
+
+* `hikari.interactions.base_interactions.ResponseType.MESSAGE_CREATE`/`4`
+* `hikari.interactions.base_interactions.ResponseType.DEFERRED_MESSAGE_CREATE`/`5`
+"""
+
+
+@attr.define(kw_only=True, weakref_slot=False)
+class PartialTextInput(messages.PartialComponent):
```

Comment:
This seems to be breaking from the normal naming conventions which would call these something along the lines of "InteractionTextInput" not "PartialTextInput"

Reply:
This was the original name but dav had an issue with it. I'm all for `InteractionTextInput` though so I'll change it back to that.